README.adoc

```diff
@@ -0,0 +1,126 @@
+
+The Hazelcast Viridian Maven Plugin is a Java development tool for testing and deploying link:https://docs.hazelcast.com/cloud/cluster-side-modules[cluster-side modules] on Serverless clusters.
+endif::[]
+
+The current version of the plugin is {page-plugin-version}.
+
+== Before you Begin
+
+Make sure that you have the following:
+
+- link:https://maven.apache.org/install.html[Maven]
+- The `JAVA_HOME` environment variable set to the path of JDK 8, 9, or 11
+
+== Usage Example
+
+ifndef::env-github[]
+[source,xml,subs="attributes+"]
+----
+<build>
+    <plugins>
+        <plugin>
+            <groupId>com.hazelcast.cloud</groupId>
+            <artifactId>hazelcast-cloud-maven-plugin</artifactId>
+            <version>{page-plugin-version}</version>
+            <configuration>
+                <clusterName>$\{clusterName}</clusterName>
+                <apiKey>$\{apiKey}</apiKey>
+                <apiSecret>$\{apiSecret}</apiSecret>
+            </configuration>
+        </plugin>
+    </plugins>
+</build>
+----
+endif::[]
+
+ifdef::env-github[]
+[source,xml,subs="attributes+"]
+----
+<build>
+    <plugins>
+        <plugin>
+            <groupId>com.hazelcast.cloud</groupId>
+            <artifactId>hazelcast-cloud-maven-plugin</artifactId>
+            <version>{page-plugin-version}</version>
+            <configuration>
+                <apiBaseUrl>https://$\{apiBaseUrl}</apiBaseUrl>
+                <clusterName>de-1234</clusterName>
+                <apiKey>$\{apiKey}</apiKey>
+                <apiSecret>$\{apiSecret}</apiSecret>
+            </configuration>
+        </plugin>
+    </plugins>
+</build>
+----
+endif::[]
+
+== Configuration
+
+[cols="1m,2a,1m"]
+|===
+| Parameter|Description| Example
+
+ifdef::env-github[]
+| apiBaseUrl
+| Base URL of the Cloud API:
+
+- *Development:* \https://dev.test.hazelcast.cloud
+- *UAT:* \https://uat.hazelcast.cloud
+- *Production:* Leave this element empty if you are using production.
+
+|\https://dev.test.hazelcast.cloud
+endif::[]
+
+| clusterName
+| Cluster name
+
+Find your cluster name in the link:https://cloud.hazelcast.com[Hazelcast Viridian console] by clicking *Connect Client*.
+| pr-1234
+
+| apiKey
+| API key
+
+Generate an API key and secret in the link:https://cloud.hazelcast.com/settings/developer[Hazelcast Viridian console].
+a|N/A
+
+| apiSecret
+| API secret
+a|N/A
+
+|===
+
+== Goals
+
+[cols="1m,2a,1a"]
+|===
+| Goal | Description | Example
+
+| deploy
+| Package your Maven project into an executable JAR file and upload that file to the connected Serverless cluster.
+|
+```bash
+mvn clean package hazelcast-cloud:deploy
+```
+
+|stream
+|Stream the latest 1,000 lines of logs from a single member in the connected Serverless cluster.
+|
+```bash
+mvn hazelcast-cloud:stream
```

Comment:
Let's remove it for now. The documentation will be already public, but this won't work until we make this feature public